app/src/main/java/com/example/android/codelabs/paging/ui/SearchRepositoriesActivity.kt

```diff
@@ -81,10 +82,15 @@ class SearchRepositoriesActivity : AppCompatActivity() {
 
     private fun initAdapter() {
         binding.list.adapter = adapter.withLoadStateHeaderAndFooter(
-                header = ReposLoadStateAdapter { adapter.retry() },
-                footer = ReposLoadStateAdapter { adapter.retry() }
+            header = ReposLoadStateAdapter { adapter.retry() },
+            footer = ReposLoadStateAdapter { adapter.retry() }
         )
         adapter.addLoadStateListener { loadState ->
+
+            // show empty list
+            val isListEmpty = loadState.refresh is LoadState.NotLoading && adapter.itemCount == 0
```

Comment:
When do you actually want the empty list to show here? Let me send you a PR for the CombinedLoadStates state machine stuff as the helper is not great in the mediator case unfortunately.

Reply:
I can also add this as a follow up if you want to merge this first.